scipy/optimize/tests/test_optimize.py

```diff
@@ -147,17 +147,6 @@ def test_bfgs(self):
         assert_allclose(self.func(params), self.func(self.solution),
                         atol=1e-6)
 
-        # Ensure that function call counts are 'known good'; these are from
-        # Scipy 0.7.0. Don't allow them to increase.
-        assert_(self.funccalls == 10, self.funccalls)
-        assert_(self.gradcalls == 8, self.gradcalls)
-
-        # Ensure that the function behaves the same; this is from Scipy 0.7.0
-        assert_allclose(self.trace[6:8],
-                        [[0, -5.25060743e-01, 4.87748473e-01],
-                         [0, -5.24885582e-01, 4.87530347e-01]],
-                        atol=1e-14, rtol=1e-7)
-
```

Comment:
This test demands the obtained solution to be *exactly* the same as the one that was obtained before. Because of modification (3) this does not happens. The function converges within fewer iterations for this problem and for a slight different point. The point for each the function converges  fulfil  the specified tolerance, as can be verified in the test: ``assert_allclose(self.func(params), self.func(self.solution), atol=1e-6)``.